### PR TITLE
Homework 9

### PR DESCRIPTION
**Pull request by volodymyrtaras**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 out
 
+target
+
```
Comment (Anderwerty) [inline .gitignore]: Remove blank lines between lines with text.

File: src/test/java/nine/MyLinkedListTest.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
+package nine;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class MyLinkedListTest {
```
Comment (Anderwerty) [inline src/test/java/nine/MyLinkedListTest.java]: You should represent test scenarios in the test-method name.

File: src/test/java/nine/MyLinkedListTest.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    public void add() {
+        MyLinkedList<Integer> myLinkedList = new MyLinkedList<>();
+        myLinkedList.add(1);
+        int expected = 1;
+        int actual = myLinkedList.size();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void remove() {
```
Comment (Anderwerty) [inline src/test/java/nine/MyLinkedListTest.java]: You should also check that you removed the right element.

File: src/test/java/nine/MyLinkedListTest.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        MyLinkedList<Integer> myLinkedList = new MyLinkedList<>();
+        myLinkedList.add(1);
+        myLinkedList.clear();
+        int expected = 0;
+        int actual = myLinkedList.size();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void size() {
```
Comment (Anderwerty) [inline src/test/java/nine/MyLinkedListTest.java]: I propose you to check two scenarios:
1) check size if the list is empty
2) check size if the list is not empty

File: src/test/java/nine/MyLinkedListTest.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        MyLinkedList myLinkedList = new MyLinkedList();
+        int expected = 0;
+        int actual = myLinkedList.size();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void get() {
+        MyLinkedList<Integer> myLinkedList = new MyLinkedList<>();
+        myLinkedList.add(1);
```
Comment (Anderwerty) [inline src/test/java/nine/MyLinkedListTest.java]: Please add more than 1 element.

File: src/test/java/nine/MyStackTest.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
+package nine;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
```
Comment (Anderwerty) [inline src/test/java/nine/MyStackTest.java]: Don't use a wild card in import.

File: src/test/java/nine/MyStackTest.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+        }
+        int expected = 10;
+        int actual = myArrayList.size();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void remove() {
+        MyStack<Integer> myStack = new MyStack<>();
+        myStack.push(1);
```
Comment (Anderwerty) [inline src/test/java/nine/MyStackTest.java]: Please push more elements.

File: src/test/java/nine/MyStackTest.java [hunk 8]
```diff
@@ -0,0 +1,65 @@
+package nine;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class MyStackTest {
```
Comment (Anderwerty) [inline src/test/java/nine/MyStackTest.java]: the same about methods names.

File: src/test/java/nine/MyStackTest.java [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+        MyStack<Integer> myStack = new MyStack<>();
+        myStack.push(1);
+        myStack.clear();
+        int expected = 0;
+        int actual = myStack.size();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void size() {
```
Comment (Anderwerty) [inline src/test/java/nine/MyStackTest.java]: Please check 2 scenarios at lest.

File: src/test/java/nine/MyStackTest.java (hunk 6 above)
Comment (volodymyrtaras) [reply src/test/java/nine/MyStackTest.java]: @Anderwerty Alexandr told us, that it is possible)) because it decrease amount of code

**Comment by volodymyrtaras** [comment]
@fixed

File: src/test/java/nine/MyStackTest.java (hunk 6 above)
Comment (Anderwerty) [reply src/test/java/nine/MyStackTest.java]: @volodymyrtaras you could get problems when you use several dependencies.
https://stackoverflow.com/questions/147454/why-is-using-a-wild-card-with-a-java-import-statement-bad 
https://myshittycode.com/2013/12/30/intellij-preventing-wildcard-imports/

File: src/test/java/nine/MyLinkedListTest.java [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+        myLinkedList.clear();
+        int expected = 0;
+        int actual = myLinkedList.size();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void gettingNotEmptySize() {
+        MyLinkedList<Integer> myLinkedList = new MyLinkedList<Integer>();
```
Comment (Anderwerty) [inline src/test/java/nine/MyLinkedListTest.java]: Please use diamond brackets. (Java 7+)

File: src/test/java/nine/MyLinkedListTest.java [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    @Test
+    public void checkEmptySize() {
+        MyLinkedList myLinkedList = new MyLinkedList();
+        int expected = 0;
+        int actual = myLinkedList.size();
+        assertEquals(expected, actual);
+    }
+
```
Comment (Anderwerty) [inline src/test/java/nine/MyLinkedListTest.java]: Remove redundant blank lines.

File: src/test/java/nine/MyArrayListTest.java [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class MyArrayListTest {
+
+    @Test
+    public void additionSizeChange() {
+        MyArrayList<String> myArrayList = new MyArrayList<>();
```
Comment (Anderwerty) [inline src/test/java/nine/MyArrayListTest.java]: Class Collections (method nCopies) would help you to make your code better.

File: src/test/java/nine/MyStackTest.java [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+        int actual = myStack.size();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void gettingNotEmptySize() {
+        MyStack<Integer> myStack = new MyStack<>();
+        myStack.push(1);
+        myStack.push(2);
+        if (myStack.size() != 0) {
```
Comment (Anderwerty) [inline src/test/java/nine/MyStackTest.java]: If myStack.size() != 0 is false assert won't work and test will be green.

**Pull request merged**